### PR TITLE
[App Service]: Fixing issues with APIs signature that changed with Track2 update

### LINKED ISSUE #18074: webapp: New Az CLI Version "2.23.0" is failing to create "az appservice domain"
> ###CLI version:

{   
**"azure-cli": "2.23.0",** 
"azure-cli-core": "2.23.0",   
"azure-cli-telemetry": "1.0.6",   
"extensions": 
{     
"ai-examples": "0.2.5",    
"ssh": "0.1.5"  
 }
 }

**Describe the bug**

The new CLI version "2.23.0" is failing to create new app service domain using the CLI command as described in this article: https://docs.microsoft.com/en-us/cli/azure/appservice/domain?view=azure-cli-latest#az_appservice_domain_create

**To Reproduce**

**Command:** 
    
az appservice domain create -g ResourceGroup --hostname mydomain.com --contact-info=@'/home/baldeep/contact_info.json' --accept-terms --subscription 703c19d3-******-*******-584ba7b4d31d

**Error:**
Command group 'appservice domain' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus The command failed with an unexpected error. Here is the traceback: check_availability() missing 1 required positional argument: 'identifier' Traceback (most recent call last):   File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 231, in invoke     cmd_result = self.invocation.execute(args)   File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 657, in execute     raise ex   File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially     results.append(self._run_job(expanded_arg, cmd_copy))   File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 691, in _run_job     result = cmd_copy(params)   File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__     return self.handler(*args, **kwargs)   File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler     return op(**command_args)   File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/appservice_domains.py", line 50, in create_domain     hostname_availability = web_client.domains.check_availability(name=hostname) TypeError: check_availability() missing 1 required positional argument: 'identifier' To open an issue, please run: 'az feedback'
Azure CLI reference types
An explanation of reference types and status
aka.ms

**Expected behavior**

Suppose to create a new domain. I think the problem is only with the new  CLI Version "2.23.0". If we downgrade it to "2.22.0" the command works fine and we are able to create the new domain with the old version.

**Additional context**

on local machine we can downgrade to old version but on Azure portal, we cannot downgrade to old version. Please fix this bug with "2.23.0" version on portal and make it more user friendly.

### PR DESCRIPTION
**Pull request by panchagnula**
Fixes #18074, #18033, #18075

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/static_sites.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def set_staticsite_function_app_settings(cmd, name, setting_pairs, resource_grou
         setting_dict[key] = value
 
     return client.create_or_update_static_site_function_app_settings(
-        resource_group_name, name, kind=None, properties=setting_dict)
+        resource_group_name, name, app_settings=setting_dict)
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/static_sites.py:123]: I saw same change recently by Jian Song.

**Review by qwordy — APPROVED** [review]

**Review by calvinsID — CHANGES_REQUESTED** [review]
Small change in `create_domain` function needed

File: src/azure-cli/azure/cli/command_modules/appservice/static_sites.py (hunk 1 above)
Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/static_sites.py:123]: Ah I see :) if that change is merged I will rebase.

File: src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py [hunk 2]
```diff
@@ -151,15 +152,16 @@ def create_domain(cmd, resource_group_name, hostname, contact_info, privacy=True
...
-        hostname_availability = web_client.domains.check_availability(name=hostname)
-    except Exception:  # pylint: disable=broad-except
-        raise CLIError("Invalid hostname: '{}'. Please enter a valid hostname".format(hostname))
+    hostname_availability = web_client.domains.check_availability(domain_identifier)
+    if not hostname_availability.available:  # api returns false
+        raise CLIError(" hostname: '{}' in not available. Please enter a valid hostname.".format(hostname))
 
     tld = '.'.join(hostname.split('.')[1:])
-    agreements = web_client.top_level_domains.list_agreements(name=tld, include_privacy=True)
+    domain_agreement_option = TopLevelDomainAgreementOption(include_privacy=True, for_transfer=True)
```
[the remark this answers is not in the document]
Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py:159]: @calvinsID fixed - please re-review when you get a chance. Thanks!

**Review by calvinsID — APPROVED** [review]

**Comment by calvinsID** [comment]
LGTM!

**Review by Juliehzl — COMMENTED** [review]
May I know why our tests doesn't catch such issue when migrating to track2 SDK? Is it because there is no test coverage for such command?

**Comment by panchagnula** [comment]
> May I know why our tests doesn't catch such issue when migrating to track2 SDK? Is it because there is no test coverage for such command?

For slots & function app - the tests were disabled , I re-enabled these with this PR.
For others there are mock tests which didn't seem to catch these errors & these cases are complex that we cannot add normal test cases for the.

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**